### DISCUSSION TITLE
x-overflow should be configurable to a DLX

### DISCUSSION THREAD

**Discussion by the1mills** [rabbitmq/rabbitmq-server#10432] [Other] [answered]
x-overflow should be configurable to a DLX

see:

https://stackoverflow.com/questions/77804007/when-using-x-max-length-and-x-overflow-how-to-send-to-dead-letter-exchange

the fact this feature doesnt exist is equally perplexing as it is depressing

**Comment by michaelklishin** [comment, marked as answer]
## Fewer emotions, more relevant details: this is an engineering discussion board, not a theater

@the1mills you won't convince anyone on this team by using terms such as "perplexing" and "depressing". You might if you make your case with fewer emotions and more details, and ideally, an executable example that would demonstrate what you are looking for.

I'm afraid I cannot decipher what exactly you asking for, leave alone why.

## `drop-head` drops messages by design (it says so right in the name)

The StackOverflow thread talks about `drop-head` and how it does not dead letter messages. But it say "drop" right in the name: it drops messages by design. A response to the SO thread correctly states that

 * `reject-publish-dlx` is the [mode](https://rabbitmq.com/maxlength.html#overflow-behaviour) you want
 * Pay attention to [unroutable messages](https://rabbitmq.com/publishers.html#unroutable) and [other aspects of publisher-side data safety](https://rabbitmq.com/publishers.html#data-safety)

I haven't seen other complaints of this kind, and DLX has been around for over a decade in RabbitMQ, with the current options about around for some six years IIRC.

Oh wait, now I remember there was this one complaint…

## "Let's all spam them with requests, that will show them this feature is in demand!"

Note that this topic has been **a subject of a spam attack** recently (a GitHub user spammed over 10 completely unrelated PRs and issues with this very link or something very similar), so I am highly sceptical that there's actually much demand for any changes in this behavior. More like a single team uses different GitHub accounts to spam us and create the illusion of this issue being in demand.

This emotional plea for (not exactly clear what specifically) to be added is exactly the strategy such teams use, we've been there before.

Now, this can be a coincidence but this does not help your case: it will take more effort and a detailed description to convince us. Our team really does not like being spammed or pressured with emotions.

**Reply by the1mills** [reply]
The functionality I am looking for

1. set x-max-length and x-oveflow to drop-head or drop-tail

2.  ̶w̶h̶e̶n̶ ̶t̶h̶e̶r̶e̶ ̶i̶s̶ ̶o̶v̶e̶r̶f̶l̶o̶w̶ ̶d̶e̶l̶e̶t̶e̶ ̶t̶h̶e̶ ̶m̶e̶s̶s̶a̶g̶e̶ ̶a̶n̶d̶ ̶s̶e̶n̶d̶ ̶t̶o̶ ̶v̶o̶i̶d̶

3. instead, when there is overflow send it to a DLX

Simple feature request imo - tell me if I am wrong, por fa

**Comment by michaelklishin** [comment]
Upvoting your own discussion makes your case look particularly weak. You really don't get it: the more you spam this team and use emotional language/illusion of interest, the less we will be inclined to look at it.

This is open source software: you very likely get it entirely for free, so all ideas and suggested improvements are up to you, even if it's a matter of convincing the team to do the work.

Throwing around words like "perplexing", "depressing", spam and upvoting your own issue are not how changes are introduced to open source software.

Sorry, I should have said that those things are perplexing and depressing to me as a long time OSS contributor.

**Reply by the1mills** [reply]
The functionality I am looking for

1. set x-max-length, and set  x-oveflow to drop-head or drop-tail

2.  ̶w̶h̶e̶n̶ ̶t̶h̶e̶r̶e̶ ̶i̶s̶ ̶o̶v̶e̶r̶f̶l̶o̶w̶ ̶d̶e̶l̶e̶t̶e̶ ̶t̶h̶e̶ ̶m̶e̶s̶s̶a̶g̶e̶ ̶a̶n̶d̶ ̶s̶e̶n̶d̶ ̶t̶o̶ ̶v̶o̶i̶d̶

3. instead, when there is overflow send it to a DLX


Simple feature request imo - tell me if I am wrong. Someone suggested `reject-publish-dlx` - but I don't think that does what I want.

**Comment by lukebakken** [comment]
@michaelklishin - @the1mills has suggested this "improvement" without making any other contributions to RabbitMQ or its associated projects, see this:

https://github.com/rabbitmq/amqp091-go/discussions/244

**Reply by the1mills** [reply]
I am a user of RMQ only, but even just as a user, I am familiar enough with the API to find missing features that many other users might want

**Reply by lukebakken** [reply]
@michaelklishin answered your request here - https://github.com/rabbitmq/rabbitmq-server/discussions/10432#discussioncomment-8276885

Use it, or contribute code, time and / or money to implement the slightly different feature you (and only you) would like.

**Reply by michaelklishin** [reply]
@the1mills you are, however, completely unfamiliar with the reasoning behind certain features, or what kind of implications the changes might entail. But you feel so super confident in every statement you make as if you were a long time contributor.

Everything is trivial to you, backwards compatibility does not exist, protocol design and common features do not exist, client libraries update themselves, team priorities do not exist either. You really know better than the core team!
